File: ClimaDaemon/CoreImplementations/Clima.Communication/Services/ServerInfoService.cs
using Clima.Communication.Messages;

namespace Clima.Communication.Services
{
    public class ServerInfoService
    {
        public ServerInfoService()
        {
        }

        public static string MessageName => "ServerInfo";

        public ServerInfoResponse GetServerVersion(ServerInfoRequest request)
        {
            return new ServerInfoResponse()
            {
                Version = "0.0.1"
            };
        }
    }
}

File: QtClient/QtClimaClient/Network/GenericServices/ServerInfoService.cpp
#include "ServerInfoService.h"

#include <Network/GenericServices/Messages/ServerInfoRequest.h>

ServerInfoService::ServerInfoService(QObject *parent) : INetworkService(parent)
{

}

void ServerInfoService::GetServerInfo()
{
    ServerInfoRequest *request = new ServerInfoRequest();
    request->method = "GetServerVersion";

    emit SendRequest(request);
}

QString ServerInfoService::ServiceName()
{
    return "ServerInfoService";
}

QList<QString> ServerInfoService::Methods()
{
    return QList<QString>();
}

void ServerInfoService::ProcessReply(const NetworkResponse &reply)
{
    qDebug()<<"ServerInfo: Process reply";
}

